src/sentry/consumers/__init__.py

```diff
@@ -578,37 +581,38 @@ def build_consumer_config(group_id: str):
             consumer_topic.value, enforce_schema, strategy_factory
         )
 
+    if stale_threshold_sec:
+        strategy_factory = DlqStaleMessagesStrategyFactoryWrapper(
+            stale_threshold_sec, strategy_factory
+        )
+
     if healthcheck_file_path is not None:
         strategy_factory = HealthcheckStrategyFactoryWrapper(
             healthcheck_file_path, strategy_factory
         )
 
     if enable_dlq and consumer_definition.get("dlq_topic"):
-        try:
-            dlq_topic = consumer_definition["dlq_topic"]
-        except KeyError as e:
-            raise click.BadParameter(
-                f"Cannot enable DLQ for consumer: {consumer_name}, no DLQ topic has been defined for it"
-            ) from e
-        try:
-            dlq_topic_defn = get_topic_definition(dlq_topic)
-            cluster_setting = dlq_topic_defn["cluster"]
-        except ValueError as e:
-            raise click.BadParameter(
-                f"Cannot enable DLQ for consumer: {consumer_name}, DLQ topic {dlq_topic} is not configured in this environment"
-            ) from e
+        dlq_topic = consumer_definition["dlq_topic"]
+    else:
+        dlq_topic = None
+
+    if stale_threshold_sec and consumer_definition.get("stale_topic"):
+        stale_topic = consumer_definition["stale_topic"]
+    else:
+        stale_topic = None
```

Comment:
nit: it doesn't really matter but this can be shortened 

```
dlq_topic = consumer_definition.get("dlq_topic") if enable_dlq else None
stale_topic = consumer_definition.get("stale_topic") if stale_threshold_sec else None
```